Data Science Project Individual Planning Stage 

We would like to know which "kinds" of players are most likely to contribute a large amount of data so that we can target those players in our recruiting efforts.

Can a player’s age predict the total amount of data they will contribute to the MineCraft server as measured by the played_hours variable in the "players.csv" dataset?

Variables: 
            1) response variable: played_hours, numeric value which represents the total play time of each player.
            2) predictor variable: age, numeric value which represents the age of the player.


In [1]:
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [2]:
players <- read_csv("players.csv")
sessions <- read_csv("sessions.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Dataset: players
Number of observations: 196 (each row represents a different player)
Number of variables: 7
Variables (Name, Type of Variable, Description): 
            1) experience, character, a player's experience level including different types of levels: Beginner, Amateur, Regular, Veteran, Pro 
            2) subscribe, logical, a TRUE / FALSE of whether the player is subscribed 
            3) hashedEmail, character, unique identifier for each player 
            4) played_hours, numeric, total hours each player has played 
            5) name, character, each player's display name
            6) gender, character, each player's gender including Male, Female, Non-binary, Prefer not to say, Other, Two-Spirited
            7) Age, numeric, each player's age in years

Potential issues: there may be duplicate names, there may be outliers in variables that could have extreme values such as played_hours

In [3]:
summary(players)

  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               

Dataset: sessions 
Number of observations: 1535 (each row represents a different player)
Number of variables: 5
Variables (Name, Type of Variable, Description):
            1) hashedEmail, character, unique identifier for each player
            2) start_time, character / datetime, starting time of the session formatted as DD/MM/YYY HH:MM
            3) end_time, character / datetime, ending time of the session formatted as DD/MM/YYY HH:MM 
            4) original_start_time, numeric, official timestamp for the start of the session (in milliseconds)
            5) original_end_time, numeric, official timestamp for the end of the session (in milliseconds)

Potential issues: the start_time and original_start_time values as well as the end_time and original_end_time have to match otherwise the data is inaccurate. The original start and end time variables have very large values so it could be more difficult to analyze the data - a potential solution could be converting the values to seconds or hours to reduce the number. 

In [4]:
summary(sessions)

 hashedEmail         start_time          end_time         original_start_time
 Length:1535        Length:1535        Length:1535        Min.   :1.712e+12  
 Class :character   Class :character   Class :character   1st Qu.:1.716e+12  
 Mode  :character   Mode  :character   Mode  :character   Median :1.719e+12  
                                                          Mean   :1.719e+12  
                                                          3rd Qu.:1.722e+12  
                                                          Max.   :1.727e+12  
                                                                             
 original_end_time  
 Min.   :1.712e+12  
 1st Qu.:1.716e+12  
 Median :1.719e+12  
 Mean   :1.719e+12  
 3rd Qu.:1.722e+12  
 Max.   :1.727e+12  
 NA's   :2          